eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTest.java

```diff
@@ -0,0 +1,325 @@
+package com.netflix.discovery.util;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTest {
```

Comment:
Bootstrapping is still very repetitive, extract common bootstrapping code for creating mock instances, application, application collections, etc and reuse these bootstrapping methods to arrange each test instead of duplicating the logic. It's important since the maintenance surface is larger, it distracts from the test logic and prone to typos.

Reply:
Apologies, but I am not sure what you mean by this.

Reply:
You actually addressed most, thanks. Left some nits but overall looks good!

Reply:
ohh ok :)